Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -9674,3 +9674,36 @@ CClientSearchLight* CStaticFunctionDefinitions::CreateSearchLight(CResource& Res
 
     return nullptr;
 }
+
+bool CStaticFunctionDefinitions::ResetSurfaceInfo()
+{
+    CSurfaceType* pSurfaceInfo = m_pClientGame->GetSurfaceInfo();
+    DWORD dwSurfaceInfo = (DWORD)pSurfaceInfo;
+
+    CSurfaceType pOriginalSurfaceInfo = m_pClientGame->GetOriginalSurfaceInfo();
+    DWORD dwOriginalSurfaceInfo = (DWORD)(&pOriginalSurfaceInfo);
+
+    short sOffset = 144;
+    short sSize = sizeof(SurfaceInfo_c) * EColSurfaceValue::SIZE;
+
+    memcpy((void*)(dwSurfaceInfo + sOffset), (void*)(dwOriginalSurfaceInfo + sOffset), sSize);
+    return true;
+}
+bool CStaticFunctionDefinitions::ResetSurfaceInfo(short sSurfaceID)
+{
+    if (sSurfaceID >= EColSurfaceValue::DEFAULT && sSurfaceID <= EColSurfaceValue::SIZE)
+    {
+        CSurfaceType pOriginalSurfaceInfo = m_pClientGame->GetOriginalSurfaceInfo();
+        CSurfaceType* pSurfaceInfo = m_pClientGame->GetSurfaceInfo();
+        DWORD dwSurfaceInfo = (DWORD)pSurfaceInfo;
+        DWORD dwOriginalSurfaceInfo = (DWORD)(&pOriginalSurfaceInfo);
+
+        short sOffset = 144 + sizeof(SurfaceInfo_c) * sSurfaceID;

```

Comment:
If **144** is the offset of `CSurfaceType::surfType` then use the **offsetof** macro from **stddef.h**:
`offsetof(CSurfaceType, surfType)`